[Bug]: EstimatePosition collateral increase estimate

### Problem encountered

slack conversation 
https://vegaprotocol.slack.com/archives/CAHA5EX0F/p1709906805588419

Ac
> In isolated margin mode: open a position so that open volume is non-zero and there are some open orders. Query the EstimatePosition with the details of that position. The collateral increase estimate should be 0. No query the EstimatePosition with a higher margin factor. The collateral increase estimate should be positive and equal to the decrease in the general account balance after margin factor is updated for the party. Now lower the margin factor below the value the test originally started with and repeat. The collateral increase estimate should be negative and equal to the decrease in the general account balance (the balance should increase). ([0013-NP-POSE-010](https://github.com/vegaprotocol/specs/blob/colosseo/non-protocol-specs/0013-NP-POSE-position-estimate.md#0013-NP-POSE-010))

The test requests for 3 estimates 
 - base estimate
 - increase margin_factor
 - decrease margin_factor
and at each estimate the test updates the traders `margin_factor` and compares actual general account diffs with the estimate

Last part of this ac is to check the collateral increase estimate equals the decrease in the general account balance however this is where there is a divergence



### Observed behaviour

the estimate returned from api and the actual diff between general account balance is significantly different

```
assert gen_acc_diff2 == int(
            best_case3
        ), f"diff of {updated_gen_acc2} - {gen_acc} does not match collateral increase estimate"
E       AssertionError: diff of 99997066940000000000000000 - 99998435940000000000000000 does not match collateral increase estimate
E       assert -1369000000000000000000 == -392000000000000000000
```

### Expected behaviour

api estimate and actual diff should be the same

### Steps to reproduce

```bash
1. run the test `test_collateral_estimate_isolated_margin` on branch `est-margin-ac`
```


### Software version

dev branch

### Failing test

test_collateral_estimate_isolated_margin

### Jenkins run

https://jenkins.vega.rocks/job/common/job/system-tests-wrapper/48220/testReport/junit/tests.margins/margins_test/test_collateral_estimate_isolated_margin/

### Configuration used

_No response_

### Relevant log output

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -60,6 +60,7 @@
 - [10055](https://github.com/vegaprotocol/vega/issues/10055) - Quitting the wallet service CLI while waiting for a password now restores the terminal capabilities.
 - [10739](https://github.com/vegaprotocol/vega/issues/10739) - Fix loading from snapshot when all markets have been settled before the loaded snapshot. 
 - [10862](https://github.com/vegaprotocol/vega/issues/10862) - Add witness configuration for node vote resend interval.
+- [10852](https://github.com/vegaprotocol/vega/issues/10852) - Fix order margin account release issue (isolated margin mode).
 
 ## 0.74.3
 

--- a/core/integration/features/margin/0019-MCAL-070.feature
+++ b/core/integration/features/margin/0019-MCAL-070.feature
@@ -190,7 +190,7 @@ Feature: Test mark price changes and closeout under isolated margin mode
 
     Then the parties should have the following account balances:
       | party  | asset | market id | margin | general | order margin |
-      | party1 | USD   | ETH/FEB23 | 71550  | 33946   | 110004       |
+      | party1 | USD   | ETH/FEB23 | 71550  | 53950   | 90000        |
 
     And the parties submit update margin mode:
       | party  | market    | margin_mode     | margin_factor | error |

--- a/core/integration/features/margin/order-margin-account-balance.feature
+++ b/core/integration/features/margin/order-margin-account-balance.feature
@@ -18,8 +18,8 @@ Feature: Test funds are transferred from general account when margin factor decr
       | party            | asset | amount       |
       | buySideProvider  | USD   | 100000000000 |
       | sellSideProvider | USD   | 100000000000 |
-      | party1           | USD   | 100000       |
-      | party2           | USD   | 100000       |
+      | party1           | USD   | 1000000      |
+      | party2           | USD   | 1000000      |
     And the parties place the following orders:
       | party            | market id | side | volume | price  | resulting trades | type       | tif     |
       | buySideProvider  | ETH/FEB23 | buy  | 10     | 14900  | 0                | TYPE_LIMIT | TIF_GTC |
@@ -33,45 +33,45 @@ Feature: Test funds are transferred from general account when margin factor decr
     When the network moves ahead "2" blocks
     Then the market data for the market "ETH/FEB23" should be:
       | mark price | trading mode            |
-      | 15900        | TRADING_MODE_CONTINUOUS |
+      | 15900      | TRADING_MODE_CONTINUOUS |
 
     When the parties submit update margin mode:
       | party  | market    | margin_mode     | margin_factor |
-      | party1 | ETH/FEB23 | isolated margin | 0.5           |
+      | party1 | ETH/FEB23 | isolated margin | 2             |
+      | party2 | ETH/FEB23 | isolated margin | 2             |
+    And the network moves ahead "2" blocks
+    Then the parties should have the following margin levels:
+      | party  | market id | maintenance | search | initial | release | margin mode     | margin factor | order |
+      | party1 | ETH/FEB23 | 16695       | 0      | 20034   | 0       | isolated margin | 2             | 96600 |
+      | party2 | ETH/FEB23 | 16695       | 0      | 20034   | 0       | isolated margin | 2             | 96600 |
+    And the parties should have the following account balances:
+      | party  | asset | market id | margin | general | order margin |
+      | party1 | USD   | ETH/FEB23 | 95400  | 808000  | 96600        |
+      | party2 | USD   | ETH/FEB23 | 95400  | 808000  | 96600        |
+
+    When the parties submit update margin mode:
+      | party  | market    | margin_mode     | margin_factor |
       | party2 | ETH/FEB23 | isolated margin | 0.5           |
     And the network moves ahead "2" blocks
     Then the parties should have the following margin levels:
       | party  | market id | maintenance | search | initial | release | margin mode     | margin factor | order |
-      | party1 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.5           | 24150 |
-      | party2 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.5           | 24150 |
+      | party1 | ETH/FEB23 | 16695       | 0      | 20034   | 0       | isolated margin | 2             | 96600 |
+      | party2 | ETH/FEB23 | 16695       | 0      | 20034   | 0       | isolated margin | 0.5           | 24150 |
     And the parties should have the following account balances:
       | party  | asset | market id | margin | general | order margin |
-      | party1 | USD   | ETH/FEB23 | 23850  | 52000   | 24150        |
-      | party2 | USD   | ETH/FEB23 | 23850  | 52000   | 24150        |
+      | party1 | USD   | ETH/FEB23 | 95400  | 808000  | 96600        |
+      | party2 | USD   | ETH/FEB23 | 23850  | 952000  | 24150        |
 
     When the parties submit update margin mode:
       | party  | market    | margin_mode     | margin_factor |
-      | party2 | ETH/FEB23 | isolated margin | 0.3           |
-    And the network moves ahead "2" blocks
-    Then the parties should have the following margin levels:
-      | party  | market id | maintenance | search | initial | release | margin mode     | margin factor | order |
-      | party1 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.5           | 24150 |
-      | party2 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.3           | 14490 |
-    And the parties should have the following account balances:
-      | party  | asset | market id | margin | general | order margin |
-      | party1 | USD   | ETH/FEB23 | 23850  | 52000   | 24150        |
-      | party2 | USD   | ETH/FEB23 | 14310  | 71200   | 14490        |
-
-    When the parties submit update margin mode:
-      | party  | market    | margin_mode     | margin_factor |
-      | party2 | ETH/FEB23 | isolated margin | 0.5           |
+      | party2 | ETH/FEB23 | isolated margin | 2             |
     And the network moves ahead "2" blocks
     # Expecting equal margin levels and balances at this stage
     Then the parties should have the following margin levels:
       | party  | market id | maintenance | search | initial | release | margin mode     | margin factor | order |
-      | party1 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.5           | 24150 |
-      | party2 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.5           | 24150 |
+      | party1 | ETH/FEB23 | 16695       | 0      | 20034   | 0       | isolated margin | 2             | 96600 |
+      | party2 | ETH/FEB23 | 16695       | 0      | 20034   | 0       | isolated margin | 2             | 96600 |
     And the parties should have the following account balances:
       | party  | asset | market id | margin | general | order margin |
-      | party1 | USD   | ETH/FEB23 | 23850  | 52000   | 24150        |
-      | party2 | USD   | ETH/FEB23 | 23850  | 52000   | 24150        |
+      | party1 | USD   | ETH/FEB23 | 95400  | 808000  | 96600        |
+      | party2 | USD   | ETH/FEB23 | 95400  | 808000  | 96600        |

--- a/core/integration/features/margin/order-margin-account-balance.feature
+++ b/core/integration/features/margin/order-margin-account-balance.feature
@@ -1,0 +1,77 @@
+Feature: Test funds are transferred from general account when margin factor decreases and released back to it when it increases
+  Background:
+    Given the following network parameters are set:
+      | name                                    | value |
+      | network.markPriceUpdateMaximumFrequency | 0s    |
+    And the liquidity monitoring parameters:
+      | name       | triggering ratio | time window | scaling factor |
+      | lqm-params | 0.00             | 24h         | 1e-9           |
+    And the simple risk model named "simple-risk-model":
+      | long | short | max move up | min move down | probability of trading |
+      | 0.1  | 0.1   | 100         | -100          | 0.2                    |
+    And the markets:
+      | id        | quote name | asset | liquidity monitoring | risk model        | margin calculator         | auction duration | fees         | price monitoring | data source config     | linear slippage factor | quadratic slippage factor | sla params      |
+      | ETH/FEB23 | ETH        | USD   | lqm-params           | simple-risk-model | default-margin-calculator | 1                | default-none | default-none     | default-eth-for-future | 0.25                   | 0                         | default-futures |
+
+  Scenario: Parties with same funds, trades, orders and margin factors must have same account balances irrespective of any interim margin factor changes
+    Given the parties deposit on asset's general account the following amount:
+      | party            | asset | amount       |
+      | buySideProvider  | USD   | 100000000000 |
+      | sellSideProvider | USD   | 100000000000 |
+      | party1           | USD   | 100000       |
+      | party2           | USD   | 100000       |
+    And the parties place the following orders:
+      | party            | market id | side | volume | price  | resulting trades | type       | tif     |
+      | buySideProvider  | ETH/FEB23 | buy  | 10     | 14900  | 0                | TYPE_LIMIT | TIF_GTC |
+      | buySideProvider  | ETH/FEB23 | buy  | 6      | 15900  | 0                | TYPE_LIMIT | TIF_GTC |
+      | party1           | ETH/FEB23 | sell | 3      | 15900  | 0                | TYPE_LIMIT | TIF_GTC |
+      | party2           | ETH/FEB23 | sell | 3      | 15900  | 0                | TYPE_LIMIT | TIF_GTC |
+      | party1           | ETH/FEB23 | sell | 3      | 16100  | 0                | TYPE_LIMIT | TIF_GTC |
+      | party2           | ETH/FEB23 | sell | 3      | 16100  | 0                | TYPE_LIMIT | TIF_GTC |
+      | sellSideProvider | ETH/FEB23 | sell | 1      | 100000 | 0                | TYPE_LIMIT | TIF_GTC |
+
+    When the network moves ahead "2" blocks
+    Then the market data for the market "ETH/FEB23" should be:
+      | mark price | trading mode            |
+      | 15900        | TRADING_MODE_CONTINUOUS |
+
+    When the parties submit update margin mode:
+      | party  | market    | margin_mode     | margin_factor |
+      | party1 | ETH/FEB23 | isolated margin | 0.5           |
+      | party2 | ETH/FEB23 | isolated margin | 0.5           |
+    And the network moves ahead "2" blocks
+    Then the parties should have the following margin levels:
+      | party  | market id | maintenance | search | initial | release | margin mode     | margin factor | order |
+      | party1 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.5           | 24150 |
+      | party2 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.5           | 24150 |
+    And the parties should have the following account balances:
+      | party  | asset | market id | margin | general | order margin |
+      | party1 | USD   | ETH/FEB23 | 23850  | 52000   | 24150        |
+      | party2 | USD   | ETH/FEB23 | 23850  | 52000   | 24150        |
+
+    When the parties submit update margin mode:
+      | party  | market    | margin_mode     | margin_factor |
+      | party2 | ETH/FEB23 | isolated margin | 0.3           |
+    And the network moves ahead "2" blocks
+    Then the parties should have the following margin levels:
+      | party  | market id | maintenance | search | initial | release | margin mode     | margin factor | order |
+      | party1 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.5           | 24150 |
+      | party2 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.3           | 14490 |
+    And the parties should have the following account balances:
+      | party  | asset | market id | margin | general | order margin |
+      | party1 | USD   | ETH/FEB23 | 23850  | 52000   | 24150        |
+      | party2 | USD   | ETH/FEB23 | 14310  | 71200   | 14490        |
+
+    When the parties submit update margin mode:
+      | party  | market    | margin_mode     | margin_factor |
+      | party2 | ETH/FEB23 | isolated margin | 0.5           |
+    And the network moves ahead "2" blocks
+    # Expecting equal margin levels and balances at this stage
+    Then the parties should have the following margin levels:
+      | party  | market id | maintenance | search | initial | release | margin mode     | margin factor | order |
+      | party1 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.5           | 24150 |
+      | party2 | ETH/FEB23 | 5370        | 0      | 6444    | 0       | isolated margin | 0.5           | 24150 |
+    And the parties should have the following account balances:
+      | party  | asset | market id | margin | general | order margin |
+      | party1 | USD   | ETH/FEB23 | 23850  | 52000   | 24150        |
+      | party2 | USD   | ETH/FEB23 | 23850  | 52000   | 24150        |

--- a/core/risk/isolated_margin.go
+++ b/core/risk/isolated_margin.go
@@ -650,7 +650,7 @@ func switchToIsolatedMargin(evt events.Margin, margin *types.MarginLevels, order
 			tp = types.TransferTypeOrderMarginLow
 		} else {
 			amt = num.UintZero().Sub(orderMarginAccountBalance, uRequireOrderMargin)
-			tp = types.TransferTypeOrderMarginLow
+			tp = types.TransferTypeOrderMarginHigh
 		}
 		riskEvents = append(riskEvents, &marginChange{
 			Margin: evt,

--- a/core/risk/isolated_margin_ex_test.go
+++ b/core/risk/isolated_margin_ex_test.go
@@ -131,7 +131,33 @@ func TestSwithToIsolatedMarginContinuous(t *testing.T) {
 	require.Equal(t, "party1", riskEvent[1].Transfer().Owner)
 	require.Equal(t, num.NewUint(300), riskEvent[0].MarginLevels().OrderMargin)
 
-	transferRecalc = requiredOrderMarginStatic.Sub(evt.OrderMarginBalance().ToDecimal())
+	// case3 - need to release from margin account and order margin account back into general account
+	evt.margin += 600
+	evt.orderMargin += 400
+	marginFactor = num.DecimalFromFloat(0.3)
+	riskEvent, err = e.SwitchToIsolatedMargin(context.Background(), evt, num.NewUint(100), num.DecimalOne(), orders, marginFactor, nil)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(riskEvent))
+	require.Equal(t, num.NewUint(310), riskEvent[0].Transfer().Amount.Amount)
+	require.Equal(t, num.NewUint(310), riskEvent[0].Transfer().MinAmount)
+	require.Equal(t, types.TransferTypeMarginHigh, riskEvent[0].Transfer().Type)
+	require.Equal(t, "party1", riskEvent[0].Transfer().Owner)
+	require.Equal(t, num.NewUint(300), riskEvent[0].MarginLevels().OrderMargin)
+
+	buyOrderInfo, sellOrderInfo = extractOrderInfo(orders)
+	requiredPositionMarginStatic, requiredOrderMarginStatic = risk.CalculateRequiredMarginInIsolatedMode(evt.size, evt.AverageEntryPrice().ToDecimal(), evt.Price().ToDecimal(), buyOrderInfo, sellOrderInfo, positionFactor, marginFactor)
+	require.True(t, !requiredPositionMarginStatic.IsZero())
+	require.True(t, !requiredOrderMarginStatic.IsZero())
+	transferRecalc = evt.MarginBalance().ToDecimal().Sub(requiredPositionMarginStatic)
+	require.True(t, riskEvent[0].Transfer().Amount.Amount.ToDecimal().Sub(transferRecalc).IsZero())
+
+	require.Equal(t, num.NewUint(100), riskEvent[1].Transfer().Amount.Amount)
+	require.Equal(t, num.NewUint(100), riskEvent[1].Transfer().MinAmount)
+	require.Equal(t, types.TransferTypeOrderMarginHigh, riskEvent[1].Transfer().Type)
+	require.Equal(t, "party1", riskEvent[1].Transfer().Owner)
+	require.Equal(t, num.NewUint(300), riskEvent[0].MarginLevels().OrderMargin)
+
+	transferRecalc = evt.OrderMarginBalance().ToDecimal().Sub(requiredOrderMarginStatic)
 	require.True(t, riskEvent[1].Transfer().Amount.Amount.ToDecimal().Sub(transferRecalc).IsZero())
 }
 

--- a/core/risk/isolated_margin_ex_test.go
+++ b/core/risk/isolated_margin_ex_test.go
@@ -130,6 +130,8 @@ func TestSwithToIsolatedMarginContinuous(t *testing.T) {
 	require.Equal(t, types.TransferTypeOrderMarginLow, riskEvent[1].Transfer().Type)
 	require.Equal(t, "party1", riskEvent[1].Transfer().Owner)
 	require.Equal(t, num.NewUint(300), riskEvent[0].MarginLevels().OrderMargin)
+	transferRecalc = requiredOrderMarginStatic.Sub(evt.OrderMarginBalance().ToDecimal())
+	require.True(t, riskEvent[1].Transfer().Amount.Amount.ToDecimal().Sub(transferRecalc).IsZero())
 
 	// case3 - need to release from margin account and order margin account back into general account
 	evt.margin += 600

